feat: Add logging format for azure devops (#457)
mkdocstrings/griffe: docs/guide/users/checking.md

The last step will fail the workflow if any breaking change is found.

### Azure Pipelines {#ci-azdo}

> Here is a quick example on how to use Griffe in an Azure Pipeline as it can be instructed to generated [task logging commands](https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#task-commands):

Reviewer: question: "as it can be instructed to generated [task logging commands" - I'm not sure to understand this sentence, could you rephrase?

Author: I hope, now it's more clear

```yaml
jobs:
  - name: check-api
    pool:
      vmImage: 'ubuntu-latest'
    steps:
      - checkout: self 
        fetchDepth: 0  # We the need the full Git history.
      - task: CmdLine@2
        inputs:
          script: pip install --user uvx
      - task: CmdLine@2
        inputs:
          # The following command will compare current changes to latest tag.
          script: | 

>             uvx griffe check --search src --format azdo your_package_name

Reviewer: question: is `uvx` really needed here? Can't we just `pip install --user griffe` and run `griffe` directly?

Author: Sure. On it

            
```

> The last step will fail the workflow if any breaking change is found.

Reviewer: ```suggestion

The last step will fail the workflow if any breaking change is found.
```

The last step will fail the workflow if any breaking change is found.

### Azure Pipelines {#ci-azdo}

> Here is a quick example on how to use Griffe in an Azure Pipeline as it is capable to use Azure DevOps [task logging commands](https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#task-commands) similar to github actions:

Reviewer: ```suggestion
Here is a quick example on how to use Griffe in an Azure Pipeline. Griffe can format its output using Azure DevOps [task logging commands](https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#task-commands), similar to GitHub Actions messages:
```

```yaml

> jobs:
>   - name: check-api
>     pool:
>       vmImage: 'ubuntu-latest'
>     steps:
>       - checkout: self 
>         fetchDepth: 0  # We the need the full Git history.
>       - task: CmdLine@2
>         inputs:
>           script: pip install --user griffe
>       - task: CmdLine@2
>         inputs:
>           # The following command will compare current changes to latest tag.
>           script: | 
>             griffe check --search src --format azdo your_package_name
>             
> ```

Reviewer: ````suggestion
```yaml
jobs:
- name: check-api
  pool:
    vmImage: 'ubuntu-latest'
  steps:
  - checkout: self 
    fetchDepth: 0  # We the need the full Git history.
  - task: CmdLine@2
    inputs:
      script: pip install --user griffe
  - task: CmdLine@2
    inputs:
      # The following command will compare current changes to latest tag.
      script: | 
        griffe check --search src --format azdo your_package_name
```
````

::warning file=src/griffe/finder.py,line=77,title=NamespacePackage.path::Attribute value was changed: `path` -> unset
```

[](){#format-azdo}

### Azure DevOps / Azure Pipelines

- **CLI**: `-f azdo`
- **API**: `check(..., style="azdo")` / `check(..., style=ExplanationStyle.AZURE_DEVOPS)`

> Similar to the github workflow syntax, griff also is capable of using the task logging command syntax used by Azure Pipelines as part of the Azure DevOps Services.

Reviewer: ```suggestion
Similarly to the GitHub workflow syntax, Griffe is also capable of using the task logging command syntax used by Azure Pipelines as part of the Azure DevOps Services.
```

::warning file=src/griffe/finder.py,line=77,title=NamespacePackage.path::Attribute value was changed: `path` -> unset
```

[](){#format-azdo}

### Azure DevOps / Azure Pipelines

- **CLI**: `-f azdo`
- **API**: `check(..., style="azdo")` / `check(..., style=ExplanationStyle.AZURE_DEVOPS)`

> Similarly to the GitHub workflow syntax, Griffe is also capable of using the task logging command syntax used by Azure Pipelines as part of the Azure DevOps Services.

Reviewer: ```suggestion
Similarly to the GitHub workflow syntax, Griffe is capable of using the task logging command syntax used by Azure Pipelines as part of the Azure DevOps Services.
```
